### PR TITLE
[#395] Fix some sonar cloud issues

### LINKED ISSUE #395: Sonar cloud issues
Fix the following issues.

https://sonarcloud.io/summary/overall?id=lifeparticle_binarytree

### PR DESCRIPTION
**Pull request by panagiotisbellias**
## Pull Request Overview

Fix some sonar issues

### Changes Introduced

Fix some sonar issues

## Linked Issue(s)

- Fixes #395 

## Testing Strategy

- [ ] Describe how you have tested these changes
- [ ] Include details of any new unit/integration tests added
- [ ] Mention any edge cases considered or testing environments (e.g., browser versions, mobile)

## Performance and Accessibility Considerations

- [ ] Confirm no significant performance drops. Use tools like Lighthouse for performance evaluation.
- [ ] Ensure new UI components are accessible (e.g., screen reader-friendly, keyboard navigable)

## Code Quality Checks

- [ ] Confirm self-review of code, including readability and maintainability
- [ ] Verify adherence to project coding standards
- [ ] Ensure proper documentation of functions and classes, if applicable

## Console Warning/Error Checks

- [ ] Confirm no new warnings or errors are showing in the console

## Configuration Changes

- [ ] If applicable, describe any changes to project configurations (e.g., `vite.config.ts`, `tsconfig.json`)

## Screenshots/Video

- [ ] Include screenshots/video demonstrating UI changes (if applicable)

## Additional Notes

- [ ] Add any other context or notes about the pull request here (optional)

### CODE REVIEW

**Comment by lifeparticle** [comment]
@panagiotisbellias Thank you for your contribution. Could you please check the build at https://app.netlify.com/sites/binarytree-dev/deploys/661fc3144c127300088d48c0 It seems to be failing.

Let me know if you have any questions.

**Comment by panagiotisbellias** [comment]
@lifeparticle fixed, but there are still some sonar issues to be fixed, I can see them in the next days in the context of a next pull request

File: ui/index.html [hunk 1]
```diff
@@ -57,6 +57,7 @@
 
 		<link rel="manifest" href="/manifest.json" />
 		<meta name="theme-color" content="#141414" />
+		<title></title>
```
Comment (lifeparticle) [inline ui/index.html:60]: @panagiotisbellias have you tested this code?

File: ui/src/components/Layouts/InputGrid/InputGrid.tsx [hunk 2]
```diff
@@ -11,13 +11,19 @@ const InputGrid: FC<InputGridProps> = ({ children, className }) => {
...
+				<Col key={getKey(child)} xs={24} sm={24} md={24} lg={12}>
 					{child}
 				</Col>
 			))}
 		</Row>
 	);
 };
 
+// Function to generate unique keys based on child content
+const getKey = (child: React.ReactNode): string => {
```
Comment (lifeparticle) [inline ui/src/components/Layouts/InputGrid/InputGrid.tsx:24]: @panagiotisbellias can you please share more about this approach. For example any articles, docs etc

Comment (lifeparticle) [reply ui/src/components/Layouts/InputGrid/InputGrid.tsx:24]: In this project we don't write helper functions inside a component.

File: ui/index.html (hunk 1 above)
Comment (panagiotisbellias) [reply ui/index.html:60]: @lifeparticle No, please confirm that I can run the application navigating to the ui directory and running `npm install`. Also, please provide the proper NodeJS version so I can test the code

Thanks,
Panagiotis

File: ui/src/components/Layouts/InputGrid/InputGrid.tsx (hunk 2 above)
Comment (panagiotisbellias) [reply ui/src/components/Layouts/InputGrid/InputGrid.tsx:24]: > @panagiotisbellias can you please share more about this approach. For example any articles, docs etc

@lifeparticle To avoid using array indices as keys, I generated unique keys for each child element using a unique identifier associated with each child element. In the getKey function, I generated the key based on the content of each child element. This ensures that each child has a unique key without relying on array indices. Probably this needs testing too when above information is provided.



> In this project we don't write helper functions inside a component.

I'm sorry, didn't notice. Please confirm that the correct location for the helper function after testing it should be `ui/src/utils/helper-functions`

Thanks,
Panagiotis

File: ui/index.html (hunk 1 above)
Comment (lifeparticle) [reply ui/index.html:60]: @panagiotisbellias please follow the steps here, https://github.com/lifeparticle/binarytree/blob/main/ui/CONTRIBUTING.md

Thanks

File: ui/src/components/Layouts/InputGrid/InputGrid.tsx (hunk 2 above)
Comment (lifeparticle) [reply ui/src/components/Layouts/InputGrid/InputGrid.tsx:24]: @panagiotisbellias Thank you for the explanation. Can you please check the official doco here, https://react.dev/learn/rendering-lists

I'm not sure if this is the right approach to generate a key.

Please put it in the same directory `./helper.ts`

Thanks

File: ui/index.html (hunk 1 above)
Comment (panagiotisbellias) [reply ui/index.html:60]: @lifeparticle I'm on it, thanks

File: ui/src/components/Layouts/InputGrid/InputGrid.tsx (hunk 2 above)
Comment (panagiotisbellias) [reply ui/src/components/Layouts/InputGrid/InputGrid.tsx:24]: @lifeparticle Indeed this is not the right approach to generate a key. This has to be done per case according to what attributes each list's object have like id, uuid etc.

I proceed with that

File: ui/index.html (hunk 1 above)
Comment (panagiotisbellias) [reply ui/index.html:60]: @lifeparticle I'm facing issues installing yarn. The v3.6.1 cannot be found either from Windows or WSL (Ubuntu 22.04)

Comment (lifeparticle) [reply ui/index.html:60]: @panagiotisbellias can you please run the following command in your Terminal:

```shell
yarn --version
```

Did you follow this [guide](https://yarnpkg.com/getting-started/install) to install yarn 3?

Comment (panagiotisbellias) [reply ui/index.html:60]: Sorry, I missed this.. Now I'm ok, moving on

Comment (panagiotisbellias) [reply ui/index.html:60]: Now my issue is this:
`PS C:\GitRepositories\personal\open-source-contribution\osc-binarytree> yarn set version 3.6.1
➤ YN0000: Retrieving https://repo.yarnpkg.com/3.6.1/packages/yarnpkg-cli/bin/yarn.js
➤ YN0000: Saving the new release in .yarn/releases/yarn-3.6.1.cjs
➤ YN0000: Done in 0s 793ms
PS C:\GitRepositories\personal\open-source-contribution\osc-binarytree> yarn --version
3.6.1
PS C:\GitRepositories\personal\open-source-contribution\osc-binarytree> yarn dev
Internal Error: root-workspace-0b6124@workspace:.: This package doesn't seem to be present in your lockfile; run "yarn install" to update the lockfile
    at V0.getCandidates (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:435:5145)
    at kf.getCandidates (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:391:1264)
    at C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:439:8033
    at df (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:390:11070)
    at ge (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:439:8013)
    at async Promise.allSettled (index 0)
    at async io (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:390:10398)
    at async C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:439:8545
    at async ti.startProgressPromise (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:390:47579)
    at async je.resolveEverything (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:439:6573)
PS C:\GitRepositories\personal\open-source-contribution\osc-binarytree> yarn install --immutable
➤ YN0070: Migrating from Yarn 1; automatically enabling the compatibility node-modules linker �

➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed

➤ YN0000: ┌ Post-resolution validation
➤ YN0000: │ @@ -1,4 +1,11 @@
➤ YN0028: │ -# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
➤ YN0028: │ -# yarn lockfile v1
➤ YN0028: │ +# This file is generated by running "yarn install" inside your project.
➤ YN0028: │ +# Manual changes might be lost - proceed with caution!
➤ YN0000: │
➤ YN0028: │ +__metadata:
➤ YN0028: │ +  version: 6
➤ YN0000: │
➤ YN0028: │ +"root-workspace-0b6124@workspace:.":
➤ YN0028: │ +  version: 0.0.0-use.local
➤ YN0028: │ +  resolution: "root-workspace-0b6124@workspace:."
➤ YN0028: │ +  languageName: unknown
➤ YN0028: │ +  linkType: soft
➤ YN0000: │
➤ YN0028: │ The lockfile would have been modified by this install, which is explicitly forbidden.
➤ YN0000: └ Completed
➤ YN0000: Failed with errors in 0s 22ms
PS C:\GitRepositories\personal\open-source-contribution\osc-binarytree> yarn dev
Internal Error: root-workspace-0b6124@workspace:.: This package doesn't seem to be present in your lockfile; run "yarn install" to update the lockfile
    at V0.getCandidates (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:435:5145)
    at kf.getCandidates (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:391:1264)
    at C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:439:8033
    at df (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:390:11070)
    at ge (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:439:8013)
    at async Promise.allSettled (index 0)
    at async io (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:390:10398)
    at async C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:439:8545
    at async ti.startProgressPromise (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:390:47579)
    at async je.resolveEverything (C:\Users\pbell\AppData\Local\node\corepack\yarn\3.6.1\yarn.js:439:6573)
PS C:\GitRepositories\personal\open-source-contribution\osc-binarytree> yarn install
➤ YN0000: ┌ Resolution step
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0000: └ Completed
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
➤ YN0000: Done in 0s 49ms
PS C:\GitRepositories\personal\open-source-contribution\osc-binarytree> yarn dev
Usage Error: Couldn't find a script named "dev".

$ yarn run [--inspect] [--inspect-brk] [-T,--top-level] [-B,--binaries-only] <scriptName> ...`

Comment (lifeparticle) [reply ui/index.html:60]: @panagiotisbellias make sure that you're inside the `ui` folder

Comment (panagiotisbellias) [reply ui/index.html:60]: Ok, now I did it with some warnings though. I see that the title is gone this way so i'm adding the correct title in the html element. This is good to be configurable though reading the title from the previous location

Comment (panagiotisbellias) [reply ui/index.html]: Hi, if more actions are needed by me please inform me or close this PR. Thanks

Comment (lifeparticle) [reply ui/index.html]: @panagiotisbellias, please pull the latest from the main and review your changes. For example, 

- Does this file need any cleanup? `ui/src/components/Layouts/PageGrid/PageGrid.tsx`
- Does the title change for every page? `ui/index.html`

Thanks

Comment (panagiotisbellias) [reply ui/index.html]: Latest changes generated merge confilcts with my side branch, so I'm closing this pull request and open a new one for the same issue when time

Thanks
